core/modules/filter/filter.pages.inc

```diff
@@ -100,6 +100,25 @@ function _filter_link_autocomplete($string) {
       $matches['/' . $path] = check_plain($result->title) . ' (' . t('Content') . ')';
       $match_count++;
     }
+    
+    				// Get file matches.
+    $results = db_select('file_managed')
+      ->fields('file_managed', array('fid', 'filename', 'uri', 'filemime', 'status'))
+      ->condition('filename', db_like($string) . '%', 'LIKE')
+      ->condition('status', 1)
+						->condition('filemime', 'image%', 'NOT LIKE')
+      ->orderBy('fid', 'DESC')
+      ->range(0, 10)
+      ->execute();
+    foreach ($results as $result) {
+      $path = $result->uri;
+      if(substr($path,0,9) == 'public://'){
```

Comment:
Maybe add a comment to explain this part. Right now, I don't understand it.

Reply:
@opi I have added some comments that I hope will help.